emu/src/render/ppu.rs

```diff
@@ -61,42 +68,74 @@ impl PixelProcessUnit {
                 todo!("BG_MODE 2 not implemented yet")
             }
             3 => {
-                // Bitmap mode
+                acquire_lock!(self.internal_memory, memory => {
+                    // Bitmap mode
+                    for y in 0..LCD_HEIGHT {
+                        for x in 0..LCD_WIDTH {
+                            let index_color = (y * LCD_WIDTH + x) * 2;
+
+                            let color_array = [
+                                memory.video_ram[index_color],
+                                memory.video_ram[index_color + 1],
+                            ];
+
+                            let color: Color = [color_array[0], color_array[1]].into();
+
+                            acquire_lock!(self.gba_lcd, gba_lcd => { gba_lcd.set_pixel(x, y, color); })
+                        }
+                    }
+                });
+            }
+            4 => {
+                // 06000000-06009FFF for Frame 0
+                // 0600A000-06013FFF for Frame 1
+                let selected_frame = acquire_lock!(self.internal_memory, memory => {
+                    memory.lcd_registers.get_frame_select()
+                });
+
                 for y in 0..LCD_HEIGHT {
                     for x in 0..LCD_WIDTH {
-                        let index_color = (y * LCD_WIDTH + x) * 2;
-                        let color: Color = [
-                            memory.video_ram[index_color],
-                            memory.video_ram[index_color + 1],
-                        ]
-                        .into();
-
-                        gba_lcd.set_pixel(x, y, color);
+                        let index_mem =
+                            (y * LCD_WIDTH + x) + (selected_frame * LCD_HEIGHT * LCD_WIDTH);
```

Comment:
I'm not sure this is correct, one frame is `0x9600` bytes long so the first one goes from `0x06000000` to `0x060095FF`, the second one starts at `0600A000`, with this `index_mem` if I were to select pixel `0;0` of Frame 1 I would have:

`0 * LCD_WIDTH + 0` + (1 * 0x9600) = 0x9600` so when accessing `video_ram` I would read from `0x6009600` which is incorrect since the Frame 1 starts from `0x600A000`

Reply:
Yep, you are right. I assumed that the frame are contiguous.

Reply:
I also noticed that for MODE_5 I did the same error ahaha

Reply:
I also noticed that for BG_MODE_5 I did the same error :( ahah

Reply:
Good catch